tools/system_libs.py

```diff
@@ -472,7 +472,7 @@ def generate_ninja(self, build_dir, libname):
     if self.deterministic_paths:
       source_dir = utils.path_from_root()
       cflags += [f'-ffile-prefix-map={source_dir}=/emsdk/emscripten',
-                 '-fdebug-compilation-dir=/emsdk/emscripten']
+                 f'-fdebug-compilation-dir={build_dir}']
```

Comment:
You mean that the DWARF entries should be like this?

```
DW_AT_name    ("/emsdk/emscripten/system/lib/compiler-rt/__trap.c")
DW_AT_comp_dir    ("/emsdk/emscripten")
```

I agree with your direction so that unnecessary build sub directories (like `cache/build/libcompiler_rt-tmp`) would be stripped out from dwarf info.